kvmagent/ansible/kvm.py

```
@@ -113,6 +113,94 @@ def update_libvirtd_config(host_post_info):
+        "  openssl req -new -x509 -days 3650 -key {ca_dir}/cakey.pem "
+        "    -out {ca_dir}/cacert.pem -subj '/O=ZStack/CN=ZStack Libvirt CA'; "
+        "fi"
+    ).format(ca_dir=ca_dir, libvirt_pki_dir=libvirt_pki_dir)
+    shell_return = os.system(command)
+    if shell_return != 0:
+        handle_ansible_info("Failed to create local CA for libvirt TLS", host_post_info, "WARNING")
+        return
+
+    # Step 2: Check if the host already has valid certs
+    command = "test -f /etc/pki/libvirt/servercert.pem && test -f /etc/pki/CA/cacert.pem"
+    (status, _) = run_remote_command(command, host_post_info, return_status=True, return_output=True)
```

Ruff (0.15.5)

[error] `run_remote_command` may be undefined, or defined from star imports

(F405)